### ISSUE TITLE
PMP TV Mode screen dimming/screen turning off while playing tv/movies

### ISSUE THREAD

**Issue by Enigma0** [plexinc/plex-media-player#507]
Compiled 2 days ago on Ubuntu Gnome 16.10.

```
            "fullscreen": true,
            "hdmi_poweron": true,
            "layout": "tv",
            "preventSystemScreensaver": true,
            "showPowerOptions": false,
            "sleepTime": 300000,
            "useOpenGL": true,
            "useSystemVideoCodecs": false,
            "webMode": "tv",
```

**Comment by wm4** [comment]
You needed xdg-screensaver for screensaver inhibiting to work.

But point taken. I just pushed a commit that uses a different method of disabling the screensaver or similar things. Please retry with git master.

**Comment by mdesdin** [comment]
@wm4 in may case (ubuntu 14 and ubuntu 16) your changes to d-bus have provoked the wrong effect, now my monitor goes to sleep when plex is playing and i am forced to press a key for the monitor to receive a signal and wake up.
i have rolled back to commit 0451a0d and all is ok again.
attaching the log of a session from yesterday in case it contains anything you can use...

[plexmediaplayer.log.txt](https://github.com/plexinc/plex-media-player/files/753265/plexmediaplayer.log.txt)

**Comment by wm4** [comment]
Because apparently our dbus power manager doesn't have code for disabling/enabling the screensaver (it was written for Embedded, where no screensaver is running).

**Comment by Enigma0** [comment]
@wm4 That difficult to add?

**Comment by wm4** [comment]
Probably not that difficult, but still needs to be written and tested by someone.

**Comment by Enigma0** [comment]
@wm4 Still not working.

**Comment by wm4** [comment]
For now I recommend passing `-DLINUX_X11POWER=on` to cmake when building PMP. This should use the xdg-screensaver script method. I don't know when I can reserve time to implement the proper dbus method.

**Comment by Enigma0** [comment]
@wm4 So unfortunately even that doesn't keep the screen on. Still turns off after 15 minutes.

**Comment by wm4** [comment]
I know of cases where the xdg-screensaver script doesn't work (even though it was made for this purpose), which might apply here.

**Comment by Enigma0** [comment]
@wm4 Can I make system changes that grant better functionality/reliability with xdg-screensaver?

**Comment by Enigma0** [comment]
Using the Caffeine extension works for the time being so long as PMP is fullscreened.

**Comment by Link-Satonaka** [comment]
I too have this problem. I can verify that I built PMP with `-DLINUX_X11POWER=on`, but xdg-screensaver does not listen.

**Comment by LongChair** [comment]
@Enigma0 : do you consider this can be closed now ?

**Comment by AeliusSaionji** [comment]
I noticed this issue still persists as of 3 months ago. I will reconfirm shortly.

**Comment by LongChair** [comment]
I was under the impression that @Enigma0 had found a fix.
But maybe i got it wrong, Please confirm if that is still the case.

**Comment by Libbum** [comment]
I'm still using wm4's interim fix for each new version&mdash;any new solution has yet to manifest for me at least. (I'm using the `plex-media-player` package from the Arch Linux user repo, which stays up to date with the release cycle AFAIK).

**Comment by Enigma0** [comment]
PMP still needs a native fix.

The workaround is to use the GNOME Extension 'Caffeine' to regulate when the screen should or shouldn't turn off but it has other issues that it causes.

**Comment by AeliusSaionji** [comment]
Yeah, #1 Caffeine is not a fix, it's a dirty hack. #2, it's now dead and the project page is steering people away from using it.

**Comment by pb1051** [comment]
how is this not resolved?

**Comment by mseeley** [comment]
We officially build and maintain macOS and Windows. We don't have Linux developers involved in the regular maintenance.

If any of you folks are handy with a knife we welcome you to come slice through this issue.

**Comment by SnaxxNZ** [comment]
I would like to chime in and say I am experiencing the same 'problem'.

I'm running Manjaro 18.04 with the KDE Desktop.
KDE Plasma Version 5.15.4
KDE Frameworks Versions 5.56.0
Qt Version 5.12.2
Kernel Version 5.0.7-1-MANJARO

After 10 mins the screen will fade to black (as per the 'Screen Energy Saving" setting).
Its almost like the system thinks it idle and doesn't realise something is playing.

I get that Linux is unsupported, but just wanted to put it out there so anyone reading this thread can see its still happening.

**Comment by AeliusSaionji** [comment]
After some investigation, I learned that many DEs inhibit sleep while audio is playing. That workaround is as good as any.

Line 11 on this script should get you started.

https://github.com/AeliusSaionji/dotfiles/blob/master/common/home/user/.local/bin/suspend-countdown.sh#L11

If someone knows a better way to inhibit sleep that doesn't involve pressing a key, let me know.

**Comment by PortStone** [comment]
+1
Manjaro 18.04 with XFCE as DE

**Comment by tsulhc** [comment]
Fedora 30 with KDE Plasma experiencing this issue

**Comment by philliplongman** [comment]
Still experiencing this on both macOS Monteray and now Raspbian Buster.
